### PR TITLE
docs: sync nextui-cli  api

### PR DESCRIPTION
**Pull request by winchesHe**
Closes # 

## 📝 Description

> Add a brief description

## ⛳️ Current behavior (updates)

> Please describe the current behavior that you are modifying

## 🚀 New behavior

> Please describe the behavior or changes this PR adds

## Summary by CodeRabbit

- **New Features**
  - Added `-p --package [string]` option to specify the package manager when initializing a new project in NextUI CLI.

- **Enhancements**
  - Updated CLI version from `v0.1.2` to `v0.2.1` across various commands.
  - Improved template selection interface for Next.js 14 with clearer descriptions.
  - Enhanced package manager selection interface.
  - Clearer upgrade confirmation display showing component version changes and required minimum versions for peer dependencies.

### CODE REVIEW

File: apps/docs/content/docs/api-references/cli-api.mdx [hunk 1]
```diff
@@ -51,13 +51,46 @@ nextui init [projectName] [options]
 ### Options
 
 - `-t --template [string]` The template to use for the new project e.g. app, pages
+- - `-p --package [string]` The package manager to use for the new project (default: `npm`)
```
Comment (wingkwong) [inline apps/docs/content/docs/api-references/cli-api.mdx:54]: Just need one `-` ?

File: apps/docs/content/docs/api-references/cli-api.mdx [hunk 2]
```diff
@@ -51,13 +51,46 @@ nextui init [projectName] [options]
...
+│  ● npm
+│  ○ yarn
+│  ○ pnpm
+│  ○ bun
+│
+◇  Template created successfully!
+│
+◇  Next steps ───────────────────╮
+│                                │
+│  cd next-app-template          │
```
Comment (wingkwong) [inline apps/docs/content/docs/api-references/cli-api.mdx:86]: should it be `my-nextui-app`?

File: apps/docs/content/docs/guide/cli.mdx [hunk 3]
```diff
@@ -83,19 +83,41 @@ nextui init my-nextui-app
...
+│  ● npm
+│  ○ yarn
+│  ○ pnpm
+│  ○ bun
+│
+◇  Template created successfully!
+│
+◇  Next steps ───────────────────╮
+│                                │
+│  cd next-app-template          │
```
Comment (wingkwong) [inline apps/docs/content/docs/guide/cli.mdx:105]: should be same as the new project name above

Comment (winchesHe) [reply apps/docs/content/docs/guide/cli.mdx]: nice catch, fixed👍

**Review by wingkwong — APPROVED** [review]

**Review by jrgarciadev — APPROVED** [review]

**Pull request merged**